app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -104,6 +104,13 @@ public void afterTextChanged(Editable s) {
             checkPermissionsAndSendLogs();
             return true;
         });
+        // Disable "useExternalStorage", "Recent upload limit" and "Use custom author name" when not logged in.
+        if (defaultKvStore.getBoolean("login_skipped", false)){
+            SwitchPreference useExternalStorage = (SwitchPreference) findPreference("useExternalStorage");
+            useExternalStorage.setEnabled(false);
+            uploadLimit.setEnabled(false);
+            useAuthorName.setEnabled(false);
+        }
     }
```

Comment:
You should disable this 3 too:
![image](https://user-images.githubusercontent.com/3127881/55862800-e0af3b80-5b81-11e9-8ff7-9e914da89543.png)

They all are related with the main activity which we don't see when we logged out.